session/session.go

```diff
@@ -176,10 +180,13 @@ func New(args ...interface{}) *Session {
 		endpointURL = DefaultEndpoint
 	}
 
+	logger := log.New(os.Stderr, "", log.LstdFlags)
+
 	sess := &Session{
 		UserName: values[keys["username"]],
 		APIKey:   values[keys["api_key"]],
 		Endpoint: endpointURL,
+		Logger:   logger,
```

Comment:
I prefer to inline this:
```go
Logger: log.New(os.Stderr, "", log.LstdFlags)
```

We also have to take care of panics when `session.New()` is not used. There should be a utility function that is used that checks for a nil Logger and automatically sets one up before sending something through the logger.